### PR TITLE
Material Block Cleanup + Block Registration Cleanup + @

### PR DESCRIPTION
**Pull request by Tictim**
## What
This PR features yet another code cleanup, this time focused on material blocks - compressed blocks, frames, and surface rocks.

## Implementation Details
Previous implementation of these blocks, `BlockCompressed`, `BlockFrame` and `BlockOre`, each used `DelayedStateBlock` for implementation. The class essentially bypassed the limit imposed by initialization order, by modifying internal variable with reflection. Since the restriction can be also bypassed by alternative approach shown in this PR, `DelayedStateBlock` is a kind of an unnecessary hack.

Additionally those three classes had very similar implementations, with most of them being identical to eachother, so I've gathered common parts to a new class `BlockMaterialBase`, and rewrote the three preexisting classes with the new class as superclass.

Two item classes, `CompressedItemBlock` and `FrameItemBlock`, were replaced by a new class `MaterialItemBlock`.

As `DelayedStateBlock` no longer has any usage after this PR, it is marked as deprecated.

Block registration stuff also received refactor: Compressed blocks, frames, and surface rocks now have separate list of blocks for registration, eliminating need for `distinct()` stream operations. Also all the block/item colors declared in `ClientProxy` have been moved directly to the usage, with better implementation. I don't know who came up with the idea of sharing one `IBlockColor`/`IItemColor` with multiple blocks and typechecking the block inside the color implementation, but it was certainly a funny one.

## Additional Information
To become true Developer of GregTech I decided to fully embrace the tradition of our ancestor - the great Archengius, creator of GT:CE Official Edition - by creating the work with same spirit and technique as him. i.e. I made an Arch commit. Below are the list of minor code refactors included in this PR.

* `MultiblockControllerBase`
  Marked `structurePattern` as nullable. Ran reformat once.
* `MetaItemBracketHandler`
  Removed unnecessary suppression.
* `BlockCompressed` / `BlockFrame` / `BlockOre`
  Removed static `getItem` util method that just delegated call to `GTUtility#getItem`.
* `BlockOre` / `OreItemBlock`
  Changed to use vanilla `setCreativeTab` instead of overriding `getCreativeTabs` manually.
* `MetaTileEntityLargeMiner` / `MetaTileEntityLargeBoiler`
  Removed returning `null` at `createStructurePattern`. It looked like there was some sort of init order problem similar to the one that led to creation of `DelayedStateBlock`, but I didn't notice any such problem both during code inspection and in-game testing.
* `MetaTileEntitySteamOven`
  Removed unused variable.
* `MetaTileEntityMultiblockTank`
  Changed nothing.

## Potential Compatibility Issues
Mods using util methods removed from this PR, like `BlockCompressed#getItem`, will be affected. Other than that I'm not sure.

### CODE REVIEW

File: src/main/java/gregtech/common/blocks/BlockFrame.java [hunk 1]
```diff
@@ -182,67 +140,68 @@ public boolean replaceWithFramedPipe(World worldIn, BlockPos pos, IBlockState st
...
                 continue;
             }
-            if (canPlaceBlockAt(worldIn, blockPos)) {
-                worldIn.setBlockState(blockPos, ((FrameItemBlock) stackInHand.getItem()).getBlockState(stackInHand));
-                SoundType type = getSoundType(stackInHand);
-                worldIn.playSound(null, pos, type.getPlaceSound(), SoundCategory.BLOCKS, (type.getVolume() + 1.0F) / 2.0F, type.getPitch() * 0.8F);
-                if (!playerIn.capabilities.isCreativeMode) {
-                    stackInHand.shrink(1);
+            if (canPlaceBlockAt(world, blockPos)) {
+                world.setBlockState(blockPos, this.getStateFromMeta(stack.getItem().getMetadata(stack.getItemDamage())));
```
Comment (TechLord22) [inline src/main/java/gregtech/common/blocks/BlockFrame.java:191]: Can this use `stack.getMetadata()` instead?

File: src/main/java/gregtech/common/blocks/BlockFrame.java [hunk 2]
```diff
@@ -303,15 +262,32 @@ public BlockFaceShape getBlockFaceShape(@Nonnull IBlockAccess worldIn, @Nonnull
         return BlockFaceShape.UNDEFINED;
     }
 
+    @Override
+    public void addInformation(ItemStack stack, @Nullable World world, List<String> tooltip, ITooltipFlag flag) {
+        if (ConfigHolder.misc.debug) {
+            tooltip.add("MetaItem Id: frame" + getGtMaterial(stack).toCamelCaseString());
```
Comment (TechLord22) [inline src/main/java/gregtech/common/blocks/BlockFrame.java:268]: Should these be localized instead of hardcoded?

File: src/main/java/gregtech/common/metatileentities/multi/MetaTileEntityLargeBoiler.java [hunk 3]
```diff
@@ -126,7 +126,7 @@ public boolean isActive() {
 
     @Override
     protected BlockPattern createStructurePattern() {
-        return boilerType == null ? null : FactoryBlockPattern.start()
+        return FactoryBlockPattern.start()
```
Comment (TechLord22) [inline src/main/java/gregtech/common/metatileentities/multi/MetaTileEntityLargeBoiler.java:129]: Instead of now NPEing here, should BoilerType be set to a default type if null?

File: src/main/java/gregtech/common/blocks/BlockFrame.java (hunk 1 above)
Comment (Tictim) [reply src/main/java/gregtech/common/blocks/BlockFrame.java:191]: `getMetadata(int)` and `getMetadata(ItemStack)` are separate method which could have different implementation, like in base `Item` class.

Although changing to ItemStack one won't change the behavior, using int one is technically correct.

File: src/main/java/gregtech/common/blocks/BlockFrame.java (hunk 2 above)
Comment (Tictim) [reply src/main/java/gregtech/common/blocks/BlockFrame.java:268]: It's only a debug info, I don't think there's need for localization.

File: src/main/java/gregtech/common/metatileentities/multi/MetaTileEntityLargeBoiler.java (hunk 3 above)
Comment (Tictim) [reply src/main/java/gregtech/common/metatileentities/multi/MetaTileEntityLargeBoiler.java:129]: Boiler type is expected to be present. If we want to support custom boiler types we should look into expanding current implementation, instead of allowing nulls.

File: src/main/java/gregtech/common/blocks/BlockMaterialBase.java [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    public Material getGtMaterial(@Nonnull IBlockState state) {
+        return state.getValue(getVariantProperty());
+    }
+
+    @Nonnull
+    public IBlockState getBlock(@Nonnull Material material) {
+        return getDefaultState().withProperty(getVariantProperty(), material);
+    }
+
+    public abstract PropertyMaterial getVariantProperty();
```
Comment (TechLord22) [inline src/main/java/gregtech/common/blocks/BlockMaterialBase.java:53]: This should probably have a `@Nonnull`.

File: src/main/java/gregtech/common/blocks/MetaBlocks.java [hunk 5]
```diff
@@ -121,11 +124,16 @@ private MetaBlocks() {}
...
-    public static final Collection<BlockOre> ORES = new ReferenceArrayList<>();
-    public static final Map<Material, BlockSurfaceRock> SURFACE_ROCK = new HashMap<>();
-    public static final Collection<BlockFluidBase> FLUID_BLOCKS = new ReferenceArrayList<>();
+    public static final Map<Material, BlockCompressed> COMPRESSED = new Object2ObjectOpenHashMap<>();
+    public static final Map<Material, BlockFrame> FRAMES = new Object2ObjectOpenHashMap<>();
+    public static final Map<Material, BlockSurfaceRock> SURFACE_ROCK = new Object2ObjectOpenHashMap<>();
+
+    public static final List<BlockCompressed> COMPRESSED_BLOCKS = new ArrayList<>();
+    public static final List<BlockFrame> FRAME_BLOCKS = new ArrayList<>();
+    public static final List<BlockSurfaceRock> SURFACE_ROCK_BLOCKS = new ArrayList<>();
```
Comment (TechLord22) [inline src/main/java/gregtech/common/blocks/MetaBlocks.java:133]: Are these lists needed? Why was the approach using `Map#values` for registering these no longer viable?

Comment (Tictim) [reply src/main/java/gregtech/common/blocks/MetaBlocks.java:133]: The maps contain multiple instance of same block, and using raw `Map#values` will make whatever logic run multiple times. Previous approach used `distinct` stream operation to prevent the scenario, which is undesirable.

File: src/main/java/gregtech/common/blocks/BlockFrame.java [hunk 6]
```diff
@@ -113,43 +100,15 @@ public int getHarvestLevel(@Nonnull IBlockState state) {
...
-        return getItem(getDefaultState().withProperty(variantProperty, material));
-    }
-
-    public IBlockState getBlock(Material material) {
-        return getDefaultState().withProperty(variantProperty, material);
-    }
-
-    public Material getGtMaterial(int meta) {
-        return variantProperty.getAllowedValues().get(meta);
+        return net.minecraft.block.material.Material.IRON;
```
Comment (ALongStringOfNumbers) [inline src/main/java/gregtech/common/blocks/BlockFrame.java:111]: Hm, why this change? I know that super should always return Material.IRON because that is set in the constructor, but I would rather just continue calling super in case we change materials on frames, or just some frames, some day

**Review by ALongStringOfNumbers — APPROVED** [review]

**Review by TechLord22 — APPROVED** [review]
